pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -159,6 +159,7 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 	err := r.update(infinispan, func() {
 		// Apply defaults and endpoint encryption settings if not already set
 		infinispan.ApplyDefaults()
+		infinispan.Spec.Affinity = podAffinity(infinispan, PodLabels(infinispan.Name))
```

Comment:
overall lgtm 
@rigazilla do we really need this defaults for any user who need this or not?

Reply:
@dmvolod yes, we're following this defaulting guidelines doc:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#defaulting

Reply:
Yes, sure.
Is it correct, that default Pod Affinity is set to `podAntiAffinity`
```
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
        - podAffinityTerm:
            labelSelector:
              matchLabels:
                app: infinispan-pod
                clusterName: example-infinispan
                infinispan_cr: example-infinispan
            topologyKey: kubernetes.io/hostname
          weight: 100
  container:
    cpu: 500m
    memory: 512Mi
```

Reply:
oh, yep. that should be fine, I've just reported the default setting we already have before to the .Spec